### DISCUSSION TITLE
Vectorization in DuckDB for queries with multiple attributes

### DISCUSSION THREAD

**Discussion by david99s** [duckdb/duckdb#2428] [Q&A] [answered]
Hello, 
I have a question regarding the understanding of the vectorized engine DuckDB is using. In your Paper "DuckDB: an Embeddable Analytical Database" it says that by default one vector includes 1024 values. Furthermore, only those columns for which are asked for in the queries are necessary, meaning only of those columns a vector gets generated, building a so called chunk. I was testing the performance for really simple queries and increased step by step the number of colums in the query, e.g. Select l_orderkey from lineitem, Select l_orderkey, l_suppkey from lineitem, select l_orderkey, l_suppkey, l_partkey from lineitem ... and so on. I was expecting the time increasing proportionally per added column. So it happened as predicted, but at a certain point (in this case when adding the 9th column) there was huge increase in time. My simple question is now, why? or did i maybe made some missings in understanding. Is it possible that DuckDB maybe decreases the number of values in a vector when a lot of different vectors for a many columns have to get generated, this would explain this situation in my opinion. 
Thank you very much in Advance!

**Comment by Mytherin** [comment, marked as answer]
The vector size does not get adapted and always stays the same. This could cause vectors to no longer fit in certain cache levels as the number of columns increases, however. For example, a single column might fit in L1, whereas after including enough columns the data might not fit in L1 anymore and needs to be moved to L2, etc.

Another possible explanation here is that not all columns in lineitem are created equal, e.g. string columns are generally more expensive to operate with than numeric columns.
